core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/UserStoreDeploymentManager.java

```diff
@@ -70,8 +70,7 @@ public void deploy(String absoluteFilePath) throws DeploymentException {
                 try {
                     primaryUSM.addSecondaryUserStoreManager(realmConfiguration, userRealm);
                 } catch (UserStoreException ex) {
-                    log.warn(ex.getMessage());
-                    throw new DeploymentException(ex.getMessage(), ex);
+                    throw new DeploymentException("Error occurred while initializing new user store manager.", ex);
```

Comment:
Shall we drop the `new` term, and mention that it's a secondary user store manager. Better to add a debug log with secondary user store manager class and the name.